Library/Homebrew/cmd/link.rb

```diff
@@ -24,15 +24,17 @@ def link
     mode.dry_run = true if ARGV.dry_run?
 
     ARGV.kegs.each do |keg|
-      if HOMEBREW_PREFIX.to_s == "/usr/local" && keg.name == "openssl"
-        opoo <<-EOS.undent
-          Refusing to link: openssl
-          Linking keg-only OpenSSL means you may end up linking against the insecure,
-          deprecated system version while using the headers from the Homebrew version.
-          Instead, pass the full include/library paths to your compiler e.g.:
-            -I#{HOMEBREW_PREFIX}/opt/openssl/include -L#{HOMEBREW_PREFIX}/opt/openssl/lib
-        EOS
-        next
+      if HOMEBREW_PREFIX.to_s == "/usr/local"
+        if keg.name.start_with?("openssl") || keg.name.start_with?("libressl")
```

Comment:
> I'd prefer we kept exempting non-`/usr/local` prefixes since there's not actually a problem there. I'd argue `openssl` shouldn't even really be `keg_only` outside of `/usr/local`, but there we go.

Maybe to provide a rationale for my thinking: Since those users have to supply custom paths to their compiler and linker invocations anyway, they could as well supply the `PREFIX/opt/openssl/{include,lib}` paths. The upside would be that their scripts/etc. will continue to work even if they decide to switch from a non-`/usr/local` to a `/usr/local` installation or share their work with other Homebrew users.

But as I said, it's just a thought. I'm fine with limiting this to `/usr/local` installations.